service-statistics-unsd/src/main/java/org/oskari/statistics/plugins/unsd/UnsdDataParser.java

```diff
@@ -105,22 +117,33 @@ public StatisticalIndicatorDataDimension getTimeperiodDimensionFromIndicatorData
         }
     }
 
-    public static void parseTimePeriod(Map<Integer, Integer> countOfAreaCodesForYear, JSONObject response) {
+    public static void parseTimePeriod(Map<Integer, Set<Integer>> geoAreaCodesForYears, JSONObject response) {
         try {
             JSONArray data = response.getJSONArray(DATA_KEY);
             for (int i = 0; i < data.length(); i++) {
                 JSONObject o = (JSONObject) data.get(i);
                 Integer year = o.getInt(TIME_PERIOD_START_KEY);
-                countOfAreaCodesForYear.merge(year, 1, Integer::sum);
+                Integer geoAreaCode = o.getInt(GEO_AREA_CODE_KEY);
+
+                if(dataObjectValueTypeIsSupported(o)) {
+                    geoAreaCodesForYears.merge(year, new HashSet<Integer>(Arrays.asList(geoAreaCode)), (oldSet,newSet) -> {
+                        return Stream.of(oldSet,newSet).flatMap(x -> x.stream()).collect(Collectors.toSet());
+                    });
```

Comment:
There's no `else` to this `if`, is this correct?

Reply:
If yes, then maybe move `Integer year = ...; and Integer geoAreaCode = ...;` inside the if-block

Reply:
I added else block with logging so implementation does not silently ignore data items with unsupported  value types.